File: src/main/java/com/wire/xenon/models/RemoteMessage.java
//
// Wire
// Copyright (C) 2016 Wire Swiss GmbH
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see http://www.gnu.org/licenses/.
//

package com.wire.xenon.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.waz.model.Messages;

import java.util.UUID;

public class RemoteMessage extends MessageBase {
    @JsonProperty
    private String assetId;
    @JsonProperty
    private String assetToken;
    @JsonProperty
    private byte[] otrKey;
    @JsonProperty
    private byte[] sha256;

    @JsonCreator
    public RemoteMessage(@JsonProperty("eventId") UUID eventId,
                         @JsonProperty("messageId") UUID messageId,
                         @JsonProperty("conversationId") UUID convId,
                         @JsonProperty("clientId") String clientId,
                         @JsonProperty("userId") UUID userId,
                         @JsonProperty("time") String time,
                         @JsonProperty("assetId") String assetId,
                         @JsonProperty("assetToken") String assetToken,
                         @JsonProperty("otrKey") byte[] otrKey,
                         @JsonProperty("sha256") byte[] sha256) {
        super(eventId, messageId, convId, clientId, userId, time);

        setAssetId(assetId);
        setAssetToken(assetToken);
        setSha256(sha256);
        setOtrKey(otrKey);
    }

    public RemoteMessage(MessageBase msg, Messages.Asset.RemoteData uploaded) {
        super(msg);

        setAssetId(uploaded.getAssetId());
        setAssetToken(uploaded.getAssetToken());
        setSha256(uploaded.getSha256().toByteArray());
        setOtrKey(uploaded.getOtrKey().toByteArray());
    }

    public String getAssetToken() {
        return assetToken;
    }

    public void setAssetToken(String assetToken) {
        this.assetToken = assetToken;
    }

    public byte[] getOtrKey() {
        return otrKey;
    }

    public void setOtrKey(byte[] otrKey) {
        this.otrKey = otrKey;
    }

    public String getAssetId() {
        return assetId;
    }

    public void setAssetId(String assetId) {
        this.assetId = assetId;
    }

    public byte[] getSha256() {
        return sha256;
    }

    public void setSha256(byte[] sha256) {
        this.sha256 = sha256;
    }
}

File: src/main/java/com/wire/kalium/models/RemoteMessage.kt
//
// Wire
// Copyright (C) 2016 Wire Swiss GmbH
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see http://www.gnu.org/licenses/.
//
package com.wire.kalium.models

import java.util.UUID
import com.waz.model.Messages.Asset.RemoteData
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonCreator

class RemoteMessage : MessageBase {
    @JsonProperty
    private var assetId: String? = null

    @JsonProperty
    private var assetToken: String? = null

    @JsonProperty
    private var otrKey: ByteArray?

    @JsonProperty
    private var sha256: ByteArray?

    @JsonCreator
    constructor(
        @JsonProperty("eventId") eventId: UUID?,
        @JsonProperty("messageId") messageId: UUID?,
        @JsonProperty("conversationId") convId: UUID?,
        @JsonProperty("clientId") clientId: String?,
        @JsonProperty("userId") userId: UUID?,
        @JsonProperty("time") time: String?,
        @JsonProperty("assetId") assetId: String?,
        @JsonProperty("assetToken") assetToken: String?,
        @JsonProperty("otrKey") otrKey: ByteArray?,
        @JsonProperty("sha256") sha256: ByteArray?
    ) : super(eventId, messageId, convId, clientId, userId, time) {
        setAssetId(assetId)
        setAssetToken(assetToken)
        setSha256(sha256)
        setOtrKey(otrKey)
    }

    constructor(msg: MessageBase?, uploaded: RemoteData?) : super(msg) {
        setAssetId(uploaded.getAssetId())
        setAssetToken(uploaded.getAssetToken())
        setSha256(uploaded.getSha256().toByteArray())
        setOtrKey(uploaded.getOtrKey().toByteArray())
    }

    fun getAssetToken(): String? {
        return assetToken
    }

    fun setAssetToken(assetToken: String?) {
        this.assetToken = assetToken
    }

    fun getOtrKey(): ByteArray? {
        return otrKey
    }

    fun setOtrKey(otrKey: ByteArray?) {
        this.otrKey = otrKey
    }

    fun getAssetId(): String? {
        return assetId
    }

    fun setAssetId(assetId: String?) {
        this.assetId = assetId
    }

    fun getSha256(): ByteArray? {
        return sha256
    }

    fun setSha256(sha256: ByteArray?) {
        this.sha256 = sha256
    }
}
